src/test/scala/com/softwaremill/realworld/articles/ArticlesEndpointsSpec.scala

```diff
@@ -80,198 +108,78 @@ object ArticlesEndpointsSpec extends ZIOSpecDefault:
+        test("return error on get") {
+          assertZIO(
+            ZIO
               .service[ArticlesEndpoints]
-              .map(_.listArticles)
+              .map(_.get)
               .flatMap { endpoint =>
                 basicRequest
-                  .get(uri"http://test.com/api/articles")
+                  .get(uri"http://test.com/api/articles/unknown-article")
                   .headers(validAuthorizationHeader())
                   .response(asJson[ArticlesList])
                   .send(backendStub(endpoint))
                   .map(_.body)
               }
-          } yield assertTrue {
-            // TODO there must be better way to implement this...
-            import com.softwaremill.realworld.common.model.UserDiff.{*, given}
-
-            val articlesList = result.toOption.get
-
-            articlesList.articlesCount == 3 &&
-            articlesList.articles.contains(
-              ArticleData(
-                "how-to-train-your-dragon",
-                "How to train your dragon",
-                "Ever wonder how?",
-                "It takes a Jacobian",
-                List("dragons", "training"),
-                Instant.ofEpochMilli(1455765776637L),
-                Instant.ofEpochMilli(1455767315824L),
-                false,
-                2,
-                ArticleAuthor("jake", Some("I work at statefarm"), Some("https://i.stack.imgur.com/xHWG8.jpg"), following = false)
-              )
-            ) &&
-            articlesList.articles.contains(
-              ArticleData(
-                "how-to-train-your-dragon-2",
-                "How to train your dragon 2",
-                "So toothless",
-                "Its a dragon",
-                List("dragons", "goats", "training"),
-                Instant.ofEpochMilli(1455765776637L),
-                Instant.ofEpochMilli(1455767315824L),
-                false,
-                1,
-                ArticleAuthor("jake", Some("I work at statefarm"), Some("https://i.stack.imgur.com/xHWG8.jpg"), following = false)
-              )
-            ) &&
-            articlesList.articles.contains(
-              ArticleData(
-                "how-to-train-your-dragon-3",
-                "How to train your dragon 3",
-                "The tagless one",
-                "Its not a dragon",
-                List(),
-                Instant.ofEpochMilli(1455765776637L),
-                Instant.ofEpochMilli(1455767315824L),
-                false,
-                0,
-                ArticleAuthor(
-                  "john",
-                  Some("I no longer work at statefarm"),
-                  Some("https://i.stack.imgur.com/xHWG8.jpg"),
-                  following = false
-                )
-              )
-            )
-          }
-        },
+          )(isLeft(equalTo(HttpError("{\"error\":\"Article with slug unknown-article doesn't exist.\"}", sttp.model.StatusCode(404)))))
+        }
+      ).provide(
+        Configuration.live,
```

Comment:
is there a way to not to repeat the same list of dependencies each time?

Reply:
we could group tests by dependencies at the top level, and then call .provide() only once for each top level suite

Reply:
or create a combined layer (as a value) and just provide it for each test?

Reply:
Can we do it in seperate PR? This one is no so related with cleaning tests and also Bartek wants to do this, so it will be better for us create new PR

Reply:
of course :)

Reply:
so please for approving this one ;)